XXX-asyncwrap-api.md

```diff
@@ -0,0 +1,309 @@
+| Title  | AsyncWrap API |
+|--------|---------------|
+| Author | @trevnorris   |
+| Status | DRAFT         |
+| Date   | 2016-04-05    |
+
+## Description
+
+Since its initial introduction with the `AsyncListener` API, `AsyncWrap` has
+slowly evolved to ensure a generalized API that would serve as a solid base for
+module authors who wished to add events to the life cycle of the event loop.
```

Comment:
I think it would be worthwhile to describe these life-cycle events at a high level.  That is, describe the user model for this API.  A lot of what I read below is tied up with implementation details around the native handles.  I think most users are concerned about when a callback was enqueued, when that callback started execution, and when that callback completed execution.